### PR TITLE
allow to use hostname in device

### PR DESCRIPTION
**Pull request by klotzt-draeger**
https://github.com/Draegerwerk/sdc11073/issues/384

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the [changelog](../CHANGELOG.md) accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/sdc11073/provider/providerimpl.py [hunk 1]
```diff
@@ -490,7 +485,10 @@ def stop_realtime_sample_loop(self):
             self.waveform_provider.stop()
 
     def get_xaddrs(self) -> list[str]:
-        addresses = self._wsdiscovery.get_active_addresses()  # these own IP addresses are currently used by discovery
+        if self._use_hostname:
+            addresses = [socket.gethostname()]
```
Comment (leon1995) [inline src/sdc11073/provider/providerimpl.py:489]: from the documentation: [gethostname()](https://docs.python.org/3/library/socket.html#socket.gethostname) doesn’t always return the fully qualified domain name; use [getfqdn()](https://docs.python.org/3/library/socket.html#socket.getfqdn) for that.

Comment (deichmab-draeger) [reply src/sdc11073/provider/providerimpl.py:489]: This PR is no branch from last master, it undoes some of the latest changes. 
A test is missing.

**Review by deichmab-draeger — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - support for python version 3.12
 - new method ContextStateTransaction.disaccociate_all
 - additional schemata for validation can be declared in SdcProviderComponents and SdcConsumerComponents
-
+- allow to use hostname instead of IP in provider
```
Comment (leon1995) [inline CHANGELOG.md:15]: add new line

File: src/sdc11073/provider/providerimpl.py [hunk 3]
```diff
@@ -490,7 +494,10 @@ def stop_realtime_sample_loop(self):
             self.waveform_provider.stop()
 
     def get_xaddrs(self) -> list[str]:
-        addresses = self._wsdiscovery.get_active_addresses()  # these own IP addresses are currently used by discovery
+        if self._use_hostname:
+            addresses = [socket.getfqdn()]
```
Comment (leon1995) [inline src/sdc11073/provider/providerimpl.py:498]: please provide a test for your new feature

**Review by leon1995 — CHANGES_REQUESTED** [review]

File: src/sdc11073/provider/providerimpl.py (hunk 1 above)
Comment (klotzt-draeger) [reply src/sdc11073/provider/providerimpl.py]: fixed (wrong cut+paste)...

File: src/sdc11073/provider/providerimpl.py (hunk 3 above)
Comment (klotzt-draeger) [reply src/sdc11073/provider/providerimpl.py:499]: done, new test+consumer/provider connection added.

**Review by deichmab-draeger — DISMISSED** [review]

File: src/sdc11073/consumer/consumerimpl.py [hunk 4]
```diff
@@ -199,6 +199,7 @@ def __init__(self, device_location: str,  # noqa: PLR0913
                  request_chunk_size: int = 0,
                  socket_timeout: int = 5,
                  force_ssl_connect: bool = False,
+                 alternative_hostname: str = None
```
Comment (leon1995) [inline src/sdc11073/consumer/consumerimpl.py:202]: typing should be `str | None`

File: src/sdc11073/consumer/consumerimpl.py [hunk 5]
```diff
@@ -340,7 +343,10 @@ def base_url(self) -> str:
         if self._http_server is None:
             return ''
         p = urlparse(self._http_server.base_url)
-        tmp = f'{p.scheme}://{self._network_adapter.ip}:{p.port}{p.path}'
+        if self._alternative_hostname:
```
Comment (leon1995) [inline src/sdc11073/consumer/consumerimpl.py:346]: can be reduced to `tmp = f'{p.scheme}://{self._alternative_hostname or self._network_adapter.ip}:{p.port}{p.path}'`

File: src/sdc11073/provider/providerimpl.py [hunk 6]
```diff
@@ -94,7 +94,9 @@ def __init__(self, ws_discovery: WsDiscoveryProtocol,
                  log_prefix: str = '',
                  default_components: SdcProviderComponents | None = None,
                  specific_components: SdcProviderComponents | None = None,
-                 chunk_size: int = 0):
+                 chunk_size: int = 0,
+                 alternative_hostname: str = None
```
Comment (leon1995) [inline src/sdc11073/provider/providerimpl.py:98]: typing `str | None`

**Review by leon1995 — CHANGES_REQUESTED** [review]

**Review by deichmab-draeger — APPROVED** [review]

File: src/sdc11073/consumer/consumerimpl.py (hunk 5 above)
Comment (leon1995) [reply src/sdc11073/consumer/consumerimpl.py:352]: will be done in next pr

File: src/sdc11073/provider/providerimpl.py (hunk 6 above)
Comment (leon1995) [reply src/sdc11073/provider/providerimpl.py:98]: will be done in next pr

**Comment by leon1995** [comment]
closed in favor of #388

**Pull request closed without merging**